tests/common/util.rs

```diff
@@ -1189,14 +1239,284 @@ impl UCommand {
+            }
+        }
+    }
+
+    /// Reads and returns the exact amount of bytes from `stdout`. If `stderr` is not redirected to
+    /// `stdout` with [`UCommand::stderr_to_stdout`] then this method may block indefinitely if the
+    /// `size` amount of bytes exceeds the amount of bytes that can be read. This method consumes
+    /// the read bytes from stdout if `stderr` is not redirected to `stdout`.
+    pub fn stdout_exact_bytes(&mut self, size: usize) -> Vec<u8> {
+        match &self.output {
+            Some(output) => output.output_exact_bytes(size),
+            None => {
+                let mut buffer = vec![0; size];
+                let stdout = self.raw.stdout.as_mut().unwrap();
+                stdout.read_exact(&mut buffer).unwrap();
+                buffer
+            }
+        }
+    }
+
+    /// Reads and returns the child's stderr as String. See also [`UChild::stderr_bytes`] for additional
+    /// information.
+    pub fn stderr(&mut self) -> String {
+        String::from_utf8(self.stderr_bytes()).unwrap()
+    }
+
+    /// Reads and returns all bytes from child's stderr. Do not call this method if the child process is
+    /// still running. If stderr is redirected to stdout with [`UCommand::stderr_to_stdout`] then always
+    /// zero bytes are returned.
+    ///
+    /// # Important
+    /// This method will block indefinitely if the child process is still running
+    pub fn stderr_bytes(&mut self) -> Vec<u8> {
+        let mut buffer = Vec::<u8>::new();
+        if self.output.is_none() {
+            let stderr = self.raw.stderr.as_mut().unwrap();
+            stderr.read_to_end(&mut buffer).unwrap();
+        }
+
+        buffer
+    }
+
+    /// Reads and returns the exact amount of bytes from stderr. If stderr is redirect to stdout with
+    /// [`UCommand::stderr_to_stdout`] then always zero bytes are returned.
+    ///
+    /// # Important
+    /// This method blocks indefinitely if the `size` amount of bytes cannot be read.
+    pub fn stderr_exact_bytes(&mut self, size: usize) -> Vec<u8> {
+        match &self.output {
+            Some(_) => vec![],
+            None => {
+                let mut buffer: Vec<u8> = vec![0; size];
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                stderr.read_exact(&mut buffer).unwrap();
+                buffer
+            }
+        }
+    }
+
+    /// Returns true if the child process is still alive and false otherwise.
+    ///
+    /// # Panics
+    /// If an error occurred during try_wait()
+    pub fn try_alive(&mut self) -> bool {
+        self.raw.try_wait().unwrap().is_none()
+    }
+
+    // TODO: improve this method and recognize if the the command already exited and if so return
+    // the code and success from the command. And maybe panic! if the process is still running and
+    // not run together with stderr_to_stdout, since this method will block indefinitely otherwise.
+    pub fn assert(&mut self) -> CmdResult {
```

Comment:
I don't like this name, because it's not asserting anything. Do we really need this method? I think with `assert_now` and `wait` we have all we need?

Reply:
yeah I think it can be removed.